platform/os_windows.go

```diff
@@ -118,3 +120,59 @@ func KillProcessByName(processName string) {
 	cmd := fmt.Sprintf("taskkill /IM %v /F", processName)
 	ExecuteCommand(cmd)
 }
+
+// CreateExtHnsNetwork creates ext HNS network if not present.
+func CreateExtHnsNetwork(createExtNetworkType string) error {
+	if len(strings.TrimSpace(createExtNetworkType)) == 0 {
+		return nil
+	}
+
+	if !strings.EqualFold(createExtNetworkType, "l2bridge") &&
+		!strings.EqualFold(createExtNetworkType, "l2tunnel") {
+		return fmt.Errorf("Invalid ext hns network type %s", createExtNetworkType)
+	}
+
+	log.Printf("[Azure CNS] CreateExtHnsNetwork")
+	extHnsNetwork, err := hcsshim.GetHNSNetworkByName("ext")
+
+	if extHnsNetwork != nil {
+		log.Printf("[Azure CNS] Found existing ext hns network with type: %s", extHnsNetwork.Type)
+		if !strings.EqualFold(createExtNetworkType, extHnsNetwork.Type) {
+			return fmt.Errorf("Network type mismatch with existing network: %s", extHnsNetwork.Type)
+		}
+
+		return nil
+	}
+
+	// create new hns network
+	log.Printf("[Azure CNS] Creating ext hns network with type %s", createExtNetworkType)
+
+	hnsNetwork := &hcsshim.HNSNetwork{
+		Name: "ext",
+		Type: createExtNetworkType,
+	}
+
+	hnsSubnet := hcsshim.Subnet{
+		AddressPrefix:  "192.168.255.0/30",
+		GatewayAddress: "192.168.255.1",
+	}
+
+	hnsNetwork.Subnets = append(hnsNetwork.Subnets, hnsSubnet)
+
+	// Marshal the request.
+	buffer, err := json.Marshal(hnsNetwork)
+	if err != nil {
+		return err
+	}
+	hnsRequest := string(buffer)
+
+	// Create the HNS network.
+	log.Printf("[Azure CNS] HNSNetworkRequest POST request:%+v", hnsRequest)
+	hnsResponse, err := hcsshim.HNSNetworkRequest("POST", "", hnsRequest)
+	log.Printf("[Azure CNS] HNSNetworkRequest POST response:%+v err:%v.", hnsResponse, err)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
no need to check if err is != nil just return err

Reply:
good catch!